### PR TITLE
New package: py-lifelines

### PR DESCRIPTION
**Pull request by aweits**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-lifelines/package.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+    - SaaS providers are interested in measuring subscriber lifetimes,
+      or time to some first action
+    - inventory stock out is a censoring event for true "demand" of a good.
+    - sociologists are interested in measuring political parties'
+      lifetimes, or relationships, or marriages
+    - A/B tests to determine how long it takes different groups to
+      perform an action.
+    *lifelines* is a pure Python implementation of the best parts of
+      survival analysis."""
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-lifelines/package.py]: `spack info` will probably break this formatting. Do we really need such a long description? A single sentence or two would probably be sufficient.

File: var/spack/repos/builtin/packages/py-lifelines/package.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    version('0.9.4', sha256='0f19a8b18ace80c231de60487b2b1a3de3eb418445c6a6d0d72c1110d860f676')
+
+    depends_on('py-setuptools', type='build')
+    depends_on('py-numpy@1.14:', type=('build', 'run'))
+    depends_on('py-scipy@1.2.0:', type=('build', 'run'))
+    depends_on('py-pandas@0.23.0:', type=('build', 'run'))
+    depends_on('py-matplotlib@3.0:', type=('build', 'run'))
+    depends_on('py-autograd@1.3:', type=('build', 'run'))
+    depends_on('py-autograd-gamma@0.3:', type=('build', 'run'))
+    depends_on('py-patsy@0.5.0:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-lifelines/package.py]: Where are all of these dependencies documented? I only see:
```
    install_requires=[                                                          
        "numpy",                                                                
        "scipy",                                                                
        "pandas>=0.18",                                                         
    ], 
```

File: var/spack/repos/builtin/packages/py-lifelines/package.py (hunk 1 above)
Comment (aweits) [reply var/spack/repos/builtin/packages/py-lifelines/package.py]: shortened

File: var/spack/repos/builtin/packages/py-lifelines/package.py (hunk 2 above)
Comment (aweits) [reply var/spack/repos/builtin/packages/py-lifelines/package.py]: oops, they came from the newest version...

File: var/spack/repos/builtin/packages/py-lifelines/package.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    *events* might refer to deaths, disease remission,
+    etc.). *lifelines* is a pure Python implementation of the best
+    parts of survival analysis."""
+
+    homepage = "https://github.com/CamDavidsonPilon/lifelines"
+    url      = "https://pypi.io/packages/source/l/lifelines/lifelines-0.25.5.tar.gz"
+
+    version('0.25.5', sha256='f24260aa371829354440dfc2c1be8d59d9e841cce7a933230213cecd67787b89')
+    version('0.9.4', sha256='0f19a8b18ace80c231de60487b2b1a3de3eb418445c6a6d0d72c1110d860f676')
+
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-lifelines/package.py:20]: Also add:
```
PYTHON_REQ = ">=3.6"                                                            
```

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**